avl_tree/AvlTree.java

```diff
@@ -0,0 +1,292 @@
+		node.right = temp;
+		sec.left = temp.right;
+		temp.right = sec;
+		node.right = temp.left;
+		temp.left = node;
+		if (node.right == null)
+			node.hRight = 0;
+		else
+			node.hRight = Math.max(node.right.hLeft, node.right.hRight) + 1;
+		if (sec.left == null)
+			sec.hLeft = 0;
+		else
+			sec.hLeft = Math.max(sec.left.hLeft, sec.left.hRight) + 1;
+		temp.hRight = Math.max(sec.hLeft, sec.hRight) + 1;
+		temp.hLeft = Math.max(node.hLeft, node.hRight) + 1;
+		return temp;
+	}
+
+	public NodeAVL rotateLL(NodeAVL node) {
+		NodeAVL temp = node.left;
+		node.left = temp.right;
+		temp.right = node;
+		if (node.left == null)
+			node.hLeft = 0;
+		else
+			node.hLeft = Math.max(node.left.hRight, node.left.hLeft) + 1;
+		temp.hRight = Math.max(node.hRight, node.hLeft) + 1;
+		return temp;
+	}
+
+	public NodeAVL rotateRR(NodeAVL node) {
+		NodeAVL temp = node.right;
+		node.right = temp.left;
+		temp.left = node;
+		if (node.right == null)
+			node.hRight = 0;
+		else
+			node.hRight = Math.max(node.right.hRight, node.right.hLeft) + 1;
+		temp.hLeft = Math.max(node.hRight, node.hLeft) + 1;
+		return temp;
+	}
+
+	public NodeAVL isRotate(NodeAVL node) { // This Method see if there is nessesity for rotation and if there is need, it do suitable rotation
+		if (node.hRight - node.hLeft >= 2) {
+			if (node.right.hRight - node.right.hLeft >= 1)
+				node = rotateRR(node);
+			else if (node.right.hRight - node.right.hLeft <= -1)
+				node = rotateRL(node);
+		} else if (node.hRight - node.hLeft <= -2) {
+			if (node.left.hRight - node.left.hLeft <= -1)
+				node = rotateLL(node);
+			else if (node.left.hRight - node.left.hLeft >= 1)
+				node = rotateLR(node);
+		}
+		return node;
+	}
+
+	public boolean isEmpty() {
+		return root == null;
+	}
+
+	public void prntIn(NodeAVL node) {
+		if (node == null)
+			return;
+		else if (node.left == null && node.right == null)
+			System.out.print(node.data + " ");
+		else {
+			prntIn(node.left);
+			System.out.print(node.data + " ");
+			prntIn(node.right);
+		}
+	}
+
+	public void del(int data) { // we call these method to delete the node if its exist.
+		if (root.data == data) {
+			NodeAVL temp = root.right;
+			if (root.left == null && root.right == null)
+				root = null;
+			else if (root.left == null)
+				root = root.right;
+			else if (temp == null)
+				root = root.left;
+			else {
+				int dta = 0;
+				if (root.right.left == null) {
+					root.right.left = root.left;
+					root = root.right;
+				} else {
+					dta = go(temp);
+					root.data = dta;
+				}
+				if (root.right == null)
+					root.hRight = 0;
+				else
+					root.hRight = Math.max(root.right.hLeft, root.right.hLeft);
+				// System.out.print(root.data);
+				root = isRotate(root);
+			}
+		} else
+			del(root, data);
+	}
+
+	private void del(NodeAVL node, int data) {	// These is the method to delete node
+		if (node.right == null && node.left == null) {
+			System.out.println("Element you wanna delete not exist");

```

Comment:
Please throw an exception from this condition rather than simply printing the `element does not exist`. Also show that this exception will be thrown by catching it in main, see other implementation of DS if you are confused what to do